fix(tutorials): continuing tutorial review MTA-1689 (#488)
scaleway/docs-content: tutorials/wireguard-mesh-vpn/index.mdx

  - compute
  - instances
dates:
  validation: 2022-03-17
  posted: 2020-07-02
---

## WireGuard - Overview

[WireGuard](https://www.wireguard.com/) is a modern VPN (Virtual Private Network) software. It is designed to be run almost anywhere, and to be cross-platform. Compared to other similar software, it is faster, more secure and simpler.

In this tutorial, we show how to use Wireguard to spin up a private mesh network to create a secure private connection between your different servers and Instances. A mesh network allows your devices to communicate between each other using a Virtual Private Network without the need for a central server:

<Lightbox src="scaleway-wireguard_mesh.webp" alt="" />

We use two Instances, one located in the `PAR-1` region, the other located in the `AMS-1` region, and create a virtual network between them.

<Message type="tip">
  

>   If you're looking to configure Wireguard as a VPN to provide secure Internet access, [check out our other Wireguard tutorial](/tutorials/install-wireguard/).

Reviewer: ```suggestion
  If you are looking to configure Wireguard as a VPN to provide secure Internet access, [check out our other Wireguard tutorial](/tutorials/install-wireguard/).
```

</Message>

<Message type="requirement">

- You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
- You have [configured your SSH Key](/console/my-project/how-to/create-ssh-key/)

> - You have two Instances running a Linux kernel ≥ 3.10, one in 

Reviewer: phrase pas finie no ?

> - You have two Instances running a Linux kernel ≥ 3.10, one in 

Author: ```suggestion
- You have two Instances running a Linux kernel ≥ 3.10.
```

- `192.168.1.1` is a randomly chosen private IP address for the VPN interface of the virtual Instances located in `PAR-1`.
- `192.168.1.2` is a randomly chosen private IP address for the VPN interface of the virtual Instances located in `AMS-1`.
- The `Endpoint` is the public IP address of every other instance.
- `52345` is a randomly chosen UDP port number used for the communication of the VPN.

### On the first Instance

<Message type="note">

In this example we assume the Instance is located in the `PAR-1` region.

</Message>

Create the file `/etc/wireguard/wg0.conf` with the following content. Replace the values in pointy brackets as necessary:

  ```
  [Interface]
  PrivateKey = <PAR-1-instance-private-key>
  Address = 192.168.1.1
  ListenPort = 52345

  [Peer]
  PublicKey = <AMS-1-instance-public-key>
  AllowedIPs = 192.168.1.2
  Endpoint = <AMS-1-instance-public-IP>:52345
  ```

### On the second Instance:

<Message type="note">

In this example we assume the Instance is located in the `AMS-1` region.

</Message>

> Create the file `/etc/wireguard/wg0.conf` with the following content. . Replace the values in pointy brackets as necessary:

Reviewer: est-ce que les . . on mettrait ps en `. . `

> Create the file `/etc/wireguard/wg0.conf` with the following content. . Replace the values in pointy brackets as necessary:

Author: ```suggestion
Create the file `/etc/wireguard/wg0.conf` with the following content. Replace the values in pointy brackets as necessary:
```
